pkg/apis/istio/v1alpha2/deepcopy.go

```diff
@@ -15,22 +15,22 @@
 package v1alpha2
 
 import (
+	"encoding/json"
 	"github.com/gogo/protobuf/proto"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 
-	"istio.io/operator/pkg/util"
 	"istio.io/pkg/log"
 )
 
 // DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
 func (m *IstioControlPlane) DeepCopyInto(out proto.Message) {
-	inj, err := util.MarshalWithJSONPB(m)
```

Comment:
I doubt we can get a correct copy without jsonpb, because the struct tags are not treated correctly. You could test it out by trying a copy with data/profiles/default.yaml and see what happens.

Reply:
I've added another test case, which reads the default.yaml file and it works fine. I could simply use UnmarshalWithJSONPB to parse the whole file, since it breaks when unmarshalling the metadata field, so the test parses only the spec, just like the existing code does.

Reply:
Ok, cool, thanks. Could you please also check that operator controller is not broken? We don't have an e2e yet for that unfortunately.

Reply:
I ran the controller locally, created the example-icp and it appeared to deploy everything properly. All the pods were up and ready. Deleting the ICP also worked properly. I haven't tested any further than that.